controller/setting_controller.go

```diff
@@ -619,9 +611,19 @@ func (sc *SettingController) updateTaintToleration() error {
 		if err != nil {
 			return err
 		}
+
 		if reflect.DeepEqual(util.TolerationListToMap(lastAppliedTolerationsList), newTolerationsMap) {
 			continue
 		}
+
+		detached, _, err := sc.ds.AreAllVolumesDetached(longhorn.DataEngineTypeAll)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check volume detachment for %v setting update", types.SettingNameTaintToleration)
+		}
+		if !detached {
+			return &types.ErrorInvalidState{Reason: fmt.Sprintf("failed to apply %v setting to Longhorn components when there are attached volumes. It will be eventually applied", types.SettingNameTaintToleration)}
+		}
+
```

Comment:
If https://github.com/longhorn/longhorn-manager/pull/2539/files#r1469726286 works, the check can be removed.
.

Reply:
Modified. Thanks